src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -4,131 +4,260 @@
+    }
 
-    @Override
-    public AbstractJob createJob(CitizenData citizen)
+    /**
+     * Assigns a field list to the field list.
+     * @param field the field to add.
+     */
+    public void addFarmerFields(Field field)
     {
-        return new JobFarmer(citizen); //TODO Implement Later
+        farmerFields.add(field);
     }
 
-    @Override
-    public void readFromNBT(NBTTagCompound compound)
+    /**
+     * Getter of the current field.
+     * @return a field object.
+     */
+    public Field getCurrentField()
     {
-        super.readFromNBT(compound);
+        return currentField;
+    }
 
-        NBTTagCompound farmerCompound = compound.getCompoundTag(TAG_FARMER);
+    /**
+     * Sets the field the farmer is currently working on.
+     * @param currentField the field to work on.
+     */
+    public void setCurrentField(final Field currentField)
+    {
+        this.currentField = currentField;
+    }
 
-        wheat = farmerCompound.getInteger(WHEAT_TAG);
-        potato = farmerCompound.getInteger(POTATO_TAG);
-        carrot = farmerCompound.getInteger(CARROT_TAG);
-        melon = farmerCompound.getInteger(MELON_TAG);
-        pumpkin = farmerCompound.getInteger(PUMPKIN_TAG);
+    /**
+     * Retrieves a random field to work on for the farmer.
+     * @return a field to work on.
+     */
+    public Field getFieldToWorkOn()
+    {
+        Collections.shuffle(farmerFields);
+        for(Field field: farmerFields)
+        {
+            if(field.needsWork())
+            {
+                currentField = field;
+                return field;
+            }
+        }
+        return null;
     }
 
     @Override
-    public void writeToNBT(NBTTagCompound compound)
+    public String getSchematicName()
     {
-        super.writeToNBT(compound);
-
-        NBTTagCompound farmerCompound = new NBTTagCompound();
+        return FARMER;
+    }
 
-        farmerCompound.setInteger(WHEAT_TAG,wheat);
-        farmerCompound.setInteger(POTATO_TAG,potato);
-        farmerCompound.setInteger(CARROT_TAG,carrot);
-        farmerCompound.setInteger(MELON_TAG,melon);
-        farmerCompound.setInteger(PUMPKIN_TAG, pumpkin);
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
 
-        compound.setTag(TAG_FARMER, farmerCompound);
+    @Override
+    public String getJobName()
+    {
+        return FARMER;
     }
 
-    public static class View extends AbstractBuildingWorker.View
+    /**
+     * Synchronize field list with colony.
+     * @param world the world the building is in.
+     */
+    public void synchWithColony(World world)
     {
-        public int wheat = 100,
-                potato = 0,
-                carrot = 0,
-                melon = 0,
-                pumpkin = 0;
+        if(!farmerFields.isEmpty())
+        {
+            ArrayList<Field> tempFields = new ArrayList<>(farmerFields);
 
+            for(Field field: tempFields)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)